src/Storage/Psr16CacheStorage.php

```diff
@@ -35,7 +35,7 @@ public function fetch($key)
      */
     public function save($key, CacheEntry $data)
     {
-        $this->cache->set($key, $data);
+        $this->cache->set($key, $data, $data->getTTL());
```

Comment:
`CacheEntry::getTTL()` can return `0` for "infinite". I think that we have to make a `if` here. What do you think? How `Psr\SimpleCache\CacheInterface` handle `0` for TTL?
https://github.com/Kevinrob/guzzle-cache-middleware/blob/master/src/CacheEntry.php#L224

You can check PSR6 for example: https://github.com/Kevinrob/guzzle-cache-middleware/blob/master/src/Storage/Psr6CacheStorage.php#L56